pulpcore/cli/common/generic.py

```diff
@@ -481,7 +482,15 @@ def _option_callback(
                     "is not valid for the {option_name} option."
                 ).format(plugin=plugin, resource_type=resource_type, option_name=param.name)
             )
-        entity_ctx: PulpEntityContext = context_class(pulp_ctx, pulp_href=pulp_href, entity=entity)
+
+        if is_command_entity:
+            entity_ctx: PulpEntityContext = init_command_entity_context(
```

Comment:
I think we should not make this a function. It'd be easier to see what happens if it just happened here.

Reply:
I wanted to make the function more readable. That is why I extracted "init_help_message" from it. I see you do not like it. ⛔

Reply:
I see the intent, but not the desired result. Sorry.
As for the help message, yes that has grown into an almost unreadable state. But I don't see you changing it. So I would urge you to not touch it here. It's easier to discuss these refactoring steps on separate PR's.